### PR TITLE
backup: add backup component and error type

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Initial the backup component and add the error type.

###  What is the type of the changes?

- New feature (a change which adds functionality)

###  How is the PR tested?

Currently, it's dead code, and will be tested in later PRs.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

File: components/backup/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[package]
+name = "backup"
+version = "0.0.1"
+edition = "2018"
+publish = false
+
+[dependencies]
+tikv = { path = "../../", default-features = false }
+kvproto = { git = "https://github.com/pingcap/kvproto.git" }
+protobuf = "2"
```
Comment (5kbpers) [inline components/backup/Cargo.toml]: Could it change to `2.8`? As same as tikv.

File: components/backup/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+[package]
+name = "backup"
+version = "0.0.1"
+edition = "2018"
```
Comment (zhangjinpeng1987) [inline components/backup/Cargo.toml]: Is this rust version?

Comment (overvenus) [reply components/backup/Cargo.toml]: Sort of, it enables new features like using crates without `extern crate ...`.

File: components/backup/src/errors.rs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+use std::{error, result};
+
+use kvproto::backup::Error as ErrorPb;
+use kvproto::errorpb::Error as RegionError;
+use kvproto::kvrpcpb::KeyError;
+use tikv::storage::kv::Error as EngineError;
+use tikv::storage::mvcc::Error as MvccError;
+use tikv::storage::txn::Error as TxnError;
+
+impl Into<ErrorPb> for Error {
```
Comment (zhangjinpeng1987) [inline components/backup/src/errors.rs]: Can we add some test for into function?

Comment (overvenus) [reply components/backup/src/errors.rs]: Tests will be added in upcoming PRs

File: components/backup/src/errors.rs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            }
+            other => {
+                err.set_msg(format!("{:?}", other));
+            }
+        }
+        err
+    }
+}
+
+quick_error! {
```
Comment (breeswish) [inline components/backup/src/errors.rs]: Prefer to use failure as new errors.

File: components/backup/src/errors.rs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            description("transaction error")
+        }
+        ClusterID(current: u64, request: u64) {
+            display("current {:?}, request {:?}", current, request)
+            description("cluster ID mismatch")
+        }
+    }
+}
+
+pub type Result<T> = result::Result<T, Error>;
```
Comment (breeswish) [inline components/backup/src/errors.rs]: This will lead to a very large `Result` even it is `Result<()>`.

File: components/backup/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![recursion_limit = "200"]
+// TODO: remove it after all code been merged.
+#![allow(unused_imports)]
+#![feature(duration_float)]
+
+#[macro_use(
```
Comment (breeswish) [inline components/backup/src/lib.rs]: tikv_alloc?

File: components/backup/src/errors.rs (hunk 5 above)
Comment (overvenus) [reply components/backup/src/errors.rs]: Good to know! Though it's okay, the performance bottleneck is in IO for this crate.

**Pull request merged**